### PR TITLE
Replace WorkflowExecutionInfo with proto equivalent

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
`persistence.WorkflowExecutionInfo` struct was replaced with `persistenceblobs.WorkflowExecutionInfo` from proto.

**Why?**
This is next step to complete `persistence.WorkflowMutableState` migration to proto to get rid of unnecessary conversion between go and proto struct.

**How did you test it?**
Run all tests.

**Potential risks**
Some edge case might not be covered by tests.

### CODE REVIEW

File: service/history/workflowTaskHandler.go [hunk 1]
```diff
@@ -641,13 +641,11 @@ func (handler *workflowTaskHandlerImpl) handleCommandContinueAsNewWorkflow(
 		return handler.failCommand(enumspb.WORKFLOW_TASK_FAILED_CAUSE_UNHANDLED_COMMAND, nil)
 	}
 
-	executionInfo := handler.mutableState.GetExecutionInfo()
```
Comment (wxing1292) [inline service/history/workflowTaskHandler.go:644]: this line should be kept so below can use `executionInfo` rather than `handler.mutableState.GetExecutionInfo()` multiple times?

File: service/history/workflowExecutionContext.go [hunk 2]
```diff
@@ -1217,11 +1217,10 @@ func (c *workflowExecutionContextImpl) enforceSizeCheck() (bool, error) {
 	}
 
 	if historySize > historySizeLimitWarn || historyCount > historyCountLimitWarn {
-		executionInfo := c.mutableState.GetExecutionInfo()
```
Comment (wxing1292) [inline service/history/workflowExecutionContext.go:1220]: keep this line so below do not have to call `c.mutableState.GetExecutionInfo()` multiple times?

File: service/history/historyBuilder.go [hunk 3]
```diff
@@ -453,11 +453,11 @@ func (b *historyBuilder) addTransientEvent(event *historypb.HistoryEvent) *histo
...
-	startRequest *historyservice.StartWorkflowExecutionRequest, previousExecution *persistence.WorkflowExecutionInfo, firstRunID, originalRunID string) *historypb.HistoryEvent {
+	startRequest *historyservice.StartWorkflowExecutionRequest, previousExecution *persistenceblobs.WorkflowExecutionInfo, previousExecutionState *persistenceblobs.WorkflowExecutionState, firstRunID, originalRunID string) *historypb.HistoryEvent {
 	var prevRunID string
 	var resetPoints *workflowpb.ResetPoints
-	if previousExecution != nil {
-		prevRunID = previousExecution.ExecutionState.RunId
+	if previousExecution != nil && previousExecutionState != nil {
+		prevRunID = previousExecutionState.RunId
 		resetPoints = previousExecution.AutoResetPoints
 	}
```
Comment (wxing1292) [inline service/history/historyBuilder.go:462]: will there be any case when `previousExecution != nil XOR previousExecutionState != nil` is true?

File: common/persistence/dataInterfaces.go [hunk 4]
```diff
@@ -463,22 +389,13 @@ type (
 		RequestCancelInfos  map[int64]*persistenceblobs.RequestCancelInfo
 		SignalInfos         map[int64]*persistenceblobs.SignalInfo
 		SignalRequestedIDs  map[string]struct{}
-		ExecutionInfo       *WorkflowExecutionInfo
-		ExecutionStats      *persistenceblobs.ExecutionStats
+		ExecutionInfo       *persistenceblobs.WorkflowExecutionInfo
+		ExecutionState      *persistenceblobs.WorkflowExecutionState
+		NextEventID         int64
```
Comment (alexshtin) [inline common/persistence/dataInterfaces.go:394]: This is core change: replace `WorkflowExecutionInfo` with `persistenceblobs.WorkflowExecutionInfo`. `NextEventID` and `ExecutionState` are not part of proto object (stored as separate columns) and `ExecutionStats` and `VersionHistories` are. Therefore, I am adding the former and removing the later.
Next step is to convert entire `WorkflowMutableState` to proto.

File: service/history/historyBuilder.go (hunk 3 above)
Comment (alexshtin) [reply service/history/historyBuilder.go]: I guess not. But because these are protos, I can remove this check all together.

File: service/history/workflowTaskHandler.go (hunk 1 above)
Comment (alexshtin) [reply service/history/workflowTaskHandler.go:644]: Initially I wanted to remove all these shortcuts, but then I realized how many of them we have and gave up. I think performance penalty is minimal but code readablity is better. You don't need to remember what is `executionInfo` and where it comes from. And this is pretty big func. Using var which is declared 100 lines above is not good.

File: service/history/workflowExecutionContext.go (hunk 2 above)
Comment (alexshtin) [reply service/history/workflowExecutionContext.go:1220]: 2 times. Two times instead of one. Do you think it worth a variable?

File: service/history/mutableStateBuilder.go [hunk 5]
```diff
@@ -3963,8 +3966,9 @@ func (e *mutableStateBuilder) CloseTransactionAsSnapshot(
 	checksum := e.generateChecksum()
 
 	workflowSnapshot := &persistence.WorkflowSnapshot{
-		ExecutionInfo:    e.executionInfo,
-		VersionHistories: e.versionHistories,
+		ExecutionInfo:  e.executionInfo,
+		ExecutionState: e.executionState,
+		NextEventID:    e.nextEventID,
```
Comment (wxing1292) [inline service/history/mutableStateBuilder.go:3971]: why the next event ID is special treated?

File: service/history/mutableStateBuilder.go [hunk 6]
```diff
@@ -124,9 +127,11 @@ type (
 		updateBufferedEvents []*historypb.HistoryEvent // buffered history events that needs to be persisted
 		clearBufferedEvents  bool                      // delete buffered events from persistence
 
-		executionInfo    *persistence.WorkflowExecutionInfo // Workflow mutable state info.
-		versionHistories *persistence.VersionHistories
-		hBuilder         *historyBuilder
+		executionInfo  *persistenceblobs.WorkflowExecutionInfo // Workflow mutable state info.
+		executionState *persistenceblobs.WorkflowExecutionState
+		nextEventID    int64
```
Comment (wxing1292) [inline service/history/mutableStateBuilder.go:132]: why the next event ID is added here?

**Review by wxing1292 — APPROVED** [review]
Please make sure this PR is forward / backward compatible

File: service/history/mutableStateBuilder.go (hunk 6 above)
Comment (alexshtin) [reply service/history/mutableStateBuilder.go:132]: It was part of old `WorkflowExecutionInfo` but it is stored as separate field in db, therefore it is not in proto struct.

**Comment by alexshtin** [comment]
> Please make sure this PR is forward / backward compatible
There are no changes in persisted objects. Only in thier in memory representations. And `mutableState` is used be history service only (no incompatible request during upgrade). So, from best of my knowledge, it is compatible.
